app/src/pages/inside/common/defectTypeTooltip/defectTypeTooltip.jsx

```diff
@@ -39,51 +42,59 @@ export class DefectTypeTooltip extends Component {
-    const defectConfig =
-      this.props.projectConfig.subTypes &&
-      this.props.projectConfig.subTypes[this.props.type.toUpperCase()];
+    const {
+      data,
+      itemId,
+      tooltipEventInfo,
+      type,
+      projectConfig,
+      intl: { formatMessage },
+    } = this.props;
+
+    const defectConfig = projectConfig.subTypes && projectConfig.subTypes[type.toUpperCase()];
 
     return (
       <div className={cx('defect-type-tooltip')}>
         {defectConfig && (
           <Fragment>
-            {this.props.type !== TO_INVESTIGATE && (
-              <DefectLink
-                itemId={this.props.itemId}
-                defects={Object.keys(this.props.data)}
-                className={cx('total-item')}
-                eventInfo={this.props.tooltipEventInfo}
-              >
-                <div className={cx('name')}>
-                  <div
-                    className={cx('circle')}
-                    style={{ backgroundColor: defectConfig[0].color }}
-                  />
-                  {formatMessage(messages[`${this.props.type}_total`])}
-                </div>
-                <span className={cx('value')}>{this.props.data.total}</span>
-              </DefectLink>
-            )}
-            {Object.keys(this.props.data).map((key) => {
-              const defectType = defectConfig.find((item) => item.locator === key);
-              return (
-                defectType && (
-                  <DefectLink
-                    key={key}
-                    itemId={this.props.itemId}
-                    defects={[key]}
-                    className={cx('item')}
-                  >
-                    <div className={cx('name')}>
-                      <div className={cx('circle')} style={{ backgroundColor: defectType.color }} />
-                      {defectType.longName}
-                    </div>
-                    <span className={cx('value')}>{this.props.data[defectType.locator]}</span>
-                  </DefectLink>
-                )
-              );
-            })}
+            <DefectLink
+              itemId={itemId}
+              defects={Object.keys(data)}
+              className={cx('total-item')}
+              eventInfo={tooltipEventInfo}
+            >
+              <div className={cx('name')}>
+                <div className={cx('circle')} style={{ backgroundColor: defectConfig[0].color }} />
+                {formatMessage(messages[`${type}_total`])}
+              </div>
+              <span className={cx('value')}>{data.total}</span>
+            </DefectLink>
+            {/**
+             * Display defect types in a proper order,
+             * the first is the system type (Product Bug, To investigate, etc.)
+             * that is way they are sorted by ID
+             * Don't display defect sub types with zero defect's amount,
+             * except system types (i.e. with index=0)
+             */
+            Object.keys(data)
```

Comment:
I think it's better to move this code as an util function.

Reply:
it could be used somewhere else?
you mean move all this sequence of map, sort, filter.. ?